source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp

```diff
@@ -280,15 +283,8 @@ void MaterialXShadingNodeImpl<BASE>::updateShader(MHWRender::MShaderInstance& sh
     documentPath.pop();
     MaterialX::FileSearchPath imageSearchPath = Plugin::instance().getResourceSearchPath(); 
     imageSearchPath.prepend(documentPath);
-        
-    // Bind environment lighting
-    // TODO: These should be options
-    static const std::string
```

Comment:
We need default values to be set on the new attributes. Otherwise everything will render black.

Reply:
I tried setting up default attribute values. The gotcha is that if the current value coincides with the default one then it's saved in the Maya scene as an empty string, so it looks impossible to represent an empty string. Is this OK in this case?

Reply:
There is some post-initialize method I believe you can use to set the value. Then it won't be the default which can stay an empty string.